### PR TITLE
[NFC][InstrProf] Rename internal `InstrProfiling` to `InstrLowerer`

### PR DESCRIPTION
**Pull request by mtrofin**
Captures its responsibility a bit better.

### CODE REVIEW

**Review by kazutakahirata — APPROVED** [review]
LGTM.  Thanks!

**Comment by mtrofin** [comment]
Ugh... @bogner just saw now @MaskRay added you. I pressed the merge button after @kazutakahirata LGTM-ed, if you have feedback on the name, please let me know and I'll happily update.

I was planning to rename the files to something like InstrProfileLoweringPass.h/.cpp, but I'll hold off if there is a longer discussion on names.

**Pull request merged**
